include/triton/Dialect/TritonGPU/Transforms/Utility.h

```diff
@@ -260,11 +260,12 @@ void replaceUsesAndPropagateType(OpBuilder &builder, Operation *oldUse,
 
 /// Replace all uses of `old` with a local load from `alloc` unless the use is a
 /// `ttg.local_alloc` with a matching shared encoding, in which case the shared
-/// memory is forwarded directly into the use.
-void replaceUsesWithLocalLoad(
-    OpBuilder &builder, OpResult old,
-    TypedValue<triton::gpu::MemDescType> alloc,
-    TypedValue<triton::gpu::AsyncTokenType> token = {});
+/// memory is forwarded directly into the use. Returns the `ttg.local_alloc` if
+/// it created one.
+std::optional<triton::gpu::LocalLoadOp>
```

Comment:
We don't need to wrap with `std::optional` again? It's very common to use `nullptr` to mean null op.